src/Notifier.cpp

```diff
@@ -163,9 +183,9 @@ size_t Notifier::ReceiveConfirmation() {
  * Reset the Shotgun-Alien queue to nullptr
  */
 void Notifier::Reset() {
-   std::lock_guard<std::mutex> guard(gLock);
    gQueue.reset(nullptr);
    gHandshakeQueue.reset(nullptr);
+   mKeepRunning.store(false);
```

Comment:
So is the purpose of this to have the Notifier in a background thread? Or is it to be accessed from two threads at one time? One thread is calling ```ReceiverConfirmation``` and another thread might want to abort the first thread so it is calling ```Reset()```. 

Please explain the use case for this

I guess I don't see the reason for the Notifier::mKeepRunning